### PR TITLE
Updates for TimeSpan and pos angle constraints

### PR DESCRIPTION
**Pull request by swalker2m**

### CODE REVIEW

File: lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala [hunk 1]
```diff
@@ -135,4 +137,29 @@ trait CoreModelDecoders {
...
+            .getOption(µs)
+            .toRight(DecodingFailure(s"Invalid TimeSpan microseconds: $µs", c.history))
+        }
+        .orElse(from("milliseconds", TimeSpan.FromMilliseconds))
+        .orElse(from("seconds", TimeSpan.FromSeconds))
+        .orElse(from("minutes", TimeSpan.FromMinutes))
+        .orElse(from("hours", TimeSpan.FromHours))
+        .orElse(from("iso", TimeSpan.FromString))
+        .orElse(DecodingFailure(s"Could not parse duration value", c.history).asLeft)
+    }
```
Comment (rpiaggio) [inline lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala:164]: If querying for units other than the canonical ones, can this result in rounding producing a different value than in the DB?

**Review by toddburnside — APPROVED** [review]
LGTM

File: lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala (hunk 1 above)
Comment (swalker2m) [reply lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala:164]: Hmm, good question.  I think it boils down to whether `TimeSpan.FromMilliseconds` etc. are valid `Format[BigDecimal, TimeSpan]` instances?  If so, then the reverse get yields the canonical `BigDecimal` form and it round trips. Certainly it passes the format tests.  The issue would be with minutes and hours, where rounding in the big decimal representation has to happen in some cases.  I _think_ at the default precision though there isn't an issue even for hours.

Comment (rpiaggio) [reply lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala:164]: Supporting only canonical units here seems enough here, which is what we've done with other similar cases (`Angle`, `Wavelength`, etc.).

I was just wondering if we were introducing possible rounding issues by supporting the rest. Should we enable all units for the other cases?

**Review by rpiaggio — APPROVED** [review]
LGTM

File: lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala (hunk 1 above)
Comment (swalker2m) [reply lucuma-schemas/src/main/scala/lucuma/schemas/decoders/CoreModelDecoders.scala:164]: I just copied this from the `lucuma-odb` schema module where I tried to supply a general purpose decoder.  I suppose we all have a pending decision on the future of `lucuma-schemas` vs. schema module/libraries per service. Until then, I don't think we gain anything from enabling decoding in all units.

**Pull request merged**